### PR TITLE
Verkholiak

### PR DESCRIPTION
**Pull request by AndriY2389**

### CODE REVIEW

File: src/main/java/com/softserve/demo/service/impl/OfferServiceImpl.java [hunk 1]
```diff
@@ -51,4 +44,14 @@ public void deleteOffer(Integer id) {
     public List<Offer> getOffersByCustomerId(Integer id) {
         return offerRepository.findAllByCustomerId(id);
     }
+
+    private Location saveLocation(final Location location) {
```
Comment (vmonchak) [inline src/main/java/com/softserve/demo/service/impl/OfferServiceImpl.java]: This method is duplicated in OrderServiceImpl. Move it to LocationService and reuse in both places

File: src/main/java/com/softserve/demo/util/validators/OrderDatesChecker.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package com.softserve.demo.util.validators;
+
+import com.softserve.demo.dto.OrderDTO;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+@Slf4j
+public class OrderDatesChecker implements ConstraintValidator<OrderDateCheck, OrderDTO> {
```
Comment (vmonchak) [inline src/main/java/com/softserve/demo/util/validators/OrderDatesChecker.java]: Offer and order date checkers are the same, so, may be, we can extract common interface for this 2 fields and have only one Checker, that accepts interface? Then we can reuse this validation in all classes, that have start/end date.
See
https://stackoverflow.com/questions/33873163/java-303-349-start-date-before-end-date-validation

File: src/main/java/com/softserve/demo/service/impl/OrderServiceImpl.java [hunk 3]
```diff
@@ -47,20 +64,48 @@ public Order createOrder(Order order) {
...
-        order.setLocation(locationFromDB);
+
+        notifyAboutOrder(order, CONTRACT_WAS_UPDATED);
+        order.setLocation(saveLocation(order.getLocation()));
         return orderRepository.save(order);
     }
 
     @Override
     public void deleteOrder(Integer id) {
         log.debug(String.format("Delete order with id: [%d]", id));
```
Comment (vmonchak) [inline src/main/java/com/softserve/demo/service/impl/OrderServiceImpl.java]: Again, use proper format for logging. 
`log.debug("Delete order with id: [{}]", id));`

File: src/main/java/com/softserve/demo/service/impl/OrderServiceImpl.java [hunk 4]
```diff
@@ -17,24 +17,41 @@
...
 public class OrderServiceImpl implements OrderService {
 
-    private OrderRepository orderRepository;
-    private LocationRepository locationRepository;
+    private static final String CONTRACT_WAS_CREATED = "New contract was created";
+    private static final String CONTRACT_WAS_UPDATED = "Contract was updated";
+    private static final String CONTRACT_WAS_DELETED = "Contract was deleted";
+    private static final String BY = " by %s %s";
+    private static final String WITH = " with %s";
+    private static final String TEXT_CREATE_CONTRACT = "Dear %s,\n" +
```
Comment (vmonchak) [inline src/main/java/com/softserve/demo/service/impl/OrderServiceImpl.java]: I would prefer to move such constants into separate class like MessageConstants or something.

**Review by SilichAlex — APPROVED** [review]

**Review by chen0899 — APPROVED** [review]

**Review by VolodymyrMoskalyuk — APPROVED** [review]

**Pull request merged**